libvast/vast/chunk.hpp

```diff
@@ -78,8 +78,8 @@ class chunk final : public caf::ref_counted {
               std::disjunction<std::is_lvalue_reference<Buffer>,
                                std::is_trivially_move_assignable<Buffer>>>>>
   static auto make(Buffer&& buffer) -> decltype(as_bytes(buffer), chunk_ptr{}) {
```

Comment:
Can we maybe document the interface requirements for this function, like `This overload can only be selected for if the argument is an rvalue-reference, has a user-defined move constructor, and an overload of 'as_bytes' exists for this type. This is intended to guard against accidental copies when calling `chunk::make()`

I'm also not really sure that it's a good idea to make the `as_bytes()` part of the signature/overload resolution, as opposed to a static_assert in the function body.

Reply:
> Can we maybe document the interface requirements for this function, like This overload can only be selected for if the argument is an rvalue-reference, has a user-defined move constructor, and an overload of 'as_bytes' exists for this type. This is intended to guard against accidental copies when calling chunk::make()`

Will add a sentence before merging.

> I'm also not really sure that it's a good idea to make the `as_bytes()` part of the signature/overload resolution, as opposed to a static_assert in the function body.

It's required so it's used for SFINAE. There's no other way to do this without `requires`, which requires C++20.